src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs

```diff
@@ -548,6 +548,25 @@ private async Task<ExtendLedgerReport> ProcessTransactions(ReadWriteDbContext db
                     ledgerTransaction.CreatedTimestamp = summary.CreatedTimestamp;
                     ledgerTransaction.NormalizedRoundTimestamp = summary.NormalizedRoundTimestamp;
                     ledgerTransaction.RawPayload = committedTransaction.LedgerTransaction.GetUnwrappedPayloadBytes();
+
+                    var eventsTuple = committedTransaction.Receipt.Events?
+                        .Select(x =>
+                        {
+                            if (x.Type.TypePointer is not CoreModel.PackageTypePointer packageTypePointer)
+                            {
+                                throw new UnreachableException("All events are expected to be package type pointer.");
```

Comment:
Wait, is that true? Why does this polymorphic type pointer even exist then?

Reply:
that's core API issue. Reported that, not sure what else can we do here, rather than fail fast.